bundles/ui/org.eclipse.smarthome.ui.classic/src/main/java/org/eclipse/smarthome/ui/classic/internal/render/GroupRenderer.java

```diff
@@ -36,16 +38,27 @@ public boolean canRender(Widget w) {
      */
     @Override
     public EList<Widget> renderWidget(Widget w, StringBuilder sb) throws RenderException {
-        String snippet = getSnippet("group");
 
-        snippet = StringUtils.replace(snippet, "%id%", itemUIRegistry.getWidgetId(w));
-        snippet = StringUtils.replace(snippet, "%icon%", escapeURLPath(itemUIRegistry.getIcon(w)));
-        snippet = StringUtils.replace(snippet, "%label%", getLabel(w));
+        if (((Group) w).isExpanded()) {
+            EList<Widget> children = itemUIRegistry.getChildren((LinkableWidget) w);
 
-        // Process the color tags
-        snippet = processColor(w, snippet);
+            TextRenderer textRenderer = new TextRenderer();
+            textRenderer.setItemUIRegistry(itemUIRegistry);
+            
+            for (Widget child : children) 
+                textRenderer.renderWidget(child, sb);            
```

Comment:
I didn't find any obvious way to have some kind of default widget provider, apart from duplicating the pagerenderer renderWidget function, I thought it was a bit out of scope of this issue, but maybe not in fact and didn't run into this exception, comprehensible by the way. What would be your opinion on this ?